flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java

```diff
@@ -305,6 +334,14 @@ public void processElement(StreamRecord<WriteResult> element) {
     this.writeResultsOfCurrentCkpt.add(element.getValue());
   }
 
+  @Override
+  public void processWatermark(Watermark mark) throws Exception {
+    super.processWatermark(mark);
+    if (mark.getTimestamp() != Watermark.MAX_WATERMARK.getTimestamp()) {
```

Comment:
why do we need to ignore the `MAX_WATERMARK`? it signals the end of input.

Reply:
As you said before, we use watermark to indicate the data completeness on the ingestion path, i think we do not need to store `MAX_WATERMARK` when flink job run in streaming-mode.
If flink job run in batch-mode, even we store one `MAX_WATERMARK`, we still can't know which partition is completed, i think in batch-mode, we can just simply rely on the scheduling system.